common/src/chain/tokens/nft.rs

```diff
@@ -17,7 +17,12 @@ use crypto::key::PublicKey;
 use serialization::{extras::non_empty_vec::DataOrNoVec, Decode, Encode};
 
 #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, serde::Serialize)]
-pub struct NftIssuance {
+pub enum NftIssuance {
```

Comment:
Doesn't this go to the blockchain? Are you sure this doesn't break serialization?

Reply:
ok I think it should be checked anyway

Reply:
I checked by taking ecnoded bytes of the block with nft issuance from master and decoding it in current branch - it works

Reply:
We have to understand why... if it was a struct, and now it's an enum, it shouldn't work, because the serialized version of the enum contains an extra byte.

Reply:
Ok I think I know why it works. Basically this diff is confusing. 
`struct NftIssuace` was renamed to `struct NftIssuanceV0`. And this struct is still a part of `OutputValue::TokenData(NftIssuanceV0)`.
Now new `enum NftIssuance` become a part of a new output type which is unrelated `TxOutput::IssueNft(NftIssuace)`. Given that the process of encoding/decoding is nested parent types just decoded proper field no matter the name.

Reply:
Just to be sure I understand: Nothing was really changed in TokenData, and we're just using a new thing in IssueNft output type that never was used before. If that's the case, then it's safe to resolve this.

Reply:
yes, NftIssuance name was just reused for a different type